### PR TITLE
[3.1.1] AWS GP3 and Throughput support

### PR DESCRIPTION
**Pull request by vasiliyskysql**

### CODE REVIEW

File: internal/provider/service_data_source.go [hunk 1]
```diff
@@ -207,6 +208,11 @@ func (d *ServiceDataSource) Schema(ctx context.Context, req datasource.SchemaReq
 						Optional:    true,
 						Description: "The number of IOPS for the storage volume. This is only applicable for io1 volumes.",
 					},
+					"throughput": schema.Int64Attribute{
+						Computed:    true,
+						Optional:    true,
+						Description: "The Throughput for the storage volume. This is only applicable for io1 volumes.",
```
Comment (thetimpanist) [inline internal/provider/service_data_source.go:214]: NB: Typo in the description. `io1` vs `gp3`

File: internal/provider/service_resource.go [hunk 2]
```diff
@@ -805,16 +824,18 @@ func (r *ServiceResource) updateAllowListState(plan *ServiceResourceModel, state
...
-			"to":        plan.Storage.ValueInt64(),
-			"iops_from": state.VolumeIOPS.ValueInt64(),
-			"iops_to":   plan.VolumeIOPS.ValueInt64(),
+			"id":              state.ID.ValueString(),
+			"from":            state.Storage.ValueInt64(),
+			"to":              plan.Storage.ValueInt64(),
+			"iops_from":       state.VolumeIOPS.ValueInt64(),
+			"iops_to":         plan.VolumeIOPS.ValueInt64(),
+			"throughput_from": state.VolumeThroughput.ValueInt64(),
+			"throughput_to":   plan.VolumeThroughput.ValueInt64(),
```
Comment (thetimpanist) [inline internal/provider/service_resource.go:835]: NB: I wonder how this behaves in practice.  If the user specifies `null` but the system defaults it to 125, what would happen on the following execution when the user still has `null` set in their terraform code.  Ideally, they wouldn't see any changes.

File: internal/provider/service_resource.go [hunk 3]
```diff
@@ -1150,15 +1171,22 @@ func (r *ServiceResource) ModifyPlan(ctx context.Context, req resource.ModifyPla
 	if plan.Provider.ValueString() == "aws" {
 		if !plan.VolumeIOPS.IsNull() && plan.VolumeType.IsNull() {
 			resp.Diagnostics.AddAttributeError(path.Root("volume_type"),
-				"volume_type is require",
-				"volume_type is required when volume_iops is set. "+
-					"Use: io1 for volume_type if volume_iops is set")
+				"volume_type is required",
+				"volume_typed is required when volume_iops is set. "+
```
Comment (thetimpanist) [inline internal/provider/service_resource.go:1175]: NB: typo `volume_typed`?

File: internal/provider/service_resource.go [hunk 4]
```diff
@@ -1150,15 +1171,22 @@ func (r *ServiceResource) ModifyPlan(ctx context.Context, req resource.ModifyPla
...
+					"Use: io1|gp2|gp3 for volume_type if volume_iops is set")
```
Comment (thetimpanist) [inline internal/provider/service_resource.go:1176]: `gp2` does not support `volume_iops`

File: internal/provider/service_resource.go [hunk 5]
```diff
@@ -1150,15 +1171,22 @@ func (r *ServiceResource) ModifyPlan(ctx context.Context, req resource.ModifyPla
...
+			return
+		}
+		if !plan.VolumeThroughput.IsNull() && plan.VolumeType.IsNull() {
+			resp.Diagnostics.AddAttributeError(path.Root("volume_type"),
+				"volume_type is required",
+				"volume_typed is required when volume_throughput is set. "+
+					"Use: io1|gp2|gp3 for volume_type if volume_throughput is set")
```
Comment (thetimpanist) [inline internal/provider/service_resource.go:1183]: `volume_throughput` is only supported on `gp3`.

File: internal/provider/service_resource.go [hunk 6]
```diff
@@ -1150,15 +1171,22 @@ func (r *ServiceResource) ModifyPlan(ctx context.Context, req resource.ModifyPla
...
 			return
 		}
-		if !plan.VolumeIOPS.IsNull() && plan.VolumeType.ValueString() != "io1" {
+		if !plan.VolumeIOPS.IsNull() && (plan.VolumeType.ValueString() != "io1" && plan.VolumeType.ValueString() != "gp2" && plan.VolumeType.ValueString() != "gp3") {
 			resp.Diagnostics.AddAttributeError(path.Root("volume_type"),
-				"volume_type must be io1 when you want to set IOPS",
-				"Use: io1 for volume_type if volume_iops is set")
+				"volume_type must be io1|gp2|gp3 when you want to set IOPS",
```
Comment (thetimpanist) [inline internal/provider/service_resource.go:1188]: Same as above, `gp2` should be removed from here.

File: internal/provider/service_resource.go (hunk 2 above)
Comment (vasiliyskysql) [reply internal/provider/service_resource.go:835]: making a change, enforcing volume_type, volume_iops, and volume_throughput to be declared. 👍

**Review by thetimpanist — APPROVED** [review]
LGTM.  Let's merge beta first and test it out before pushing the production PR.

**Pull request merged**